mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -56,62 +75,76 @@ class RenderList {
 
   RenderList(this.elements);
 
+  static Map<String, List<Asset>> _groupAssets(
+    List<Asset> assets,
+    GroupAssetsBy groupBy,
+  ) {
+    assets.sortByCompare<DateTime>(
+      (e) => e.createdAt,
+      (a, b) => b.compareTo(a),
+    );
+
+    if (groupBy == GroupAssetsBy.yearMonthDay) {
+      return assets.groupListsBy(
+        (element) => DateFormat('y-MM-dd').format(element.createdAt.toLocal()),
+      );
+    } else if (groupBy == GroupAssetsBy.yearMonth) {
+      return assets.groupListsBy(
+        (element) => DateFormat('y-MM').format(element.createdAt.toLocal()),
+      );
+    }
+
+    return {};
+  }
+
   static Future<RenderList> _processAssetGroupData(
     _AssetGroupsToRenderListComputeParameters data,
   ) async {
     final monthFormat = DateFormat(data.monthFormat);
     final dayFormatSameYear = DateFormat(data.dayFormat);
     final dayFormatOtherYear = DateFormat(data.dayFormatYear);
-    final groups = data.groups;
-    final perRow = data.perRow;
+    final allAssets = data.assets;
+    final perRow = data.layout.perRow;
+    final dynamicLayout = data.layout.dynamicLayout;
+    final groupBy = data.layout.groupBy;
 
     List<RenderAssetGridElement> elements = [];
     DateTime? lastDate;
 
+    final groups = _groupAssets(allAssets, groupBy);
+
     groups.forEach((groupName, assets) {
       try {
-        final date = DateTime.parse(groupName);
-
-        if (lastDate == null || lastDate!.month != date.month) {
-          // Month title
-
-          var monthTitleText = groupName;
-
-          var groupDate = DateTime.tryParse(groupName);
-          if (groupDate != null) {
-            monthTitleText = monthFormat.format(groupDate);
-          } else {
-            log.severe("Failed to format date for day title: $groupName");
-          }
+        final date = assets[0].createdAt;
```

Comment:
Is this a safe access? Maybe guard this against `assets.isNotEmpty`

Reply:
It' safe. Otherwise there would not be a group at all.